### PR TITLE
Task/65 approver should be able to decline po

### PR DESCRIPTION
**Pull request by mllrdev-jerry-bayoneta**

### CODE REVIEW

File: src/core/enums/status.enum.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 enum PurchaseStatus {
-    PENDING = 'pending',
-    APPROVED = 'approved',
-    DECLINED = 'declined',
-    CLOSE = 'close'
+    PENDING = 'PENDING',
+    APPROVED = 'APPROVED',
+    DENIED = 'DENIED',
+    CLOSE = 'CLOSE'
```
Comment (mllrdev-carlo-medel) [inline src/core/enums/status.enum.ts]:
```suggestion
    CLOSED= 'CLOSED'
```

File: src/core/services/order.service.ts [hunk 2]
```diff
@@ -49,6 +50,20 @@ const create = async (order: IPurchaseOrder) => {
...
+    data: statusUpdate,
+    method: 'PATCH',
+    url: 'order/updateOrder'
+  })
+    .then( response => {
+      console.log(response.data.message);
+    })
+    .catch( error => {
+      console.log(error)
+    })
```
Comment (mllrdev-carlo-medel) [inline src/core/services/order.service.ts]: Maybe return something here like a boolean if the updating is success or not and handle it accordingly in the view.

File: src/pages/Create.vue [hunk 3]
```diff
@@ -154,6 +178,10 @@ export default defineComponent({
       orderService.requestor.create(this.orderDetail);
       router.push("/home");
     },
+    declinePurchaseOrder() {
+      this.orderStatusUpdate.orderDetails.status = PurchaseStatus.DENIED;
+      orderService.approver.purchaseStatusUpdate(this.orderStatusUpdate);
```
Comment (mllrdev-carlo-medel) [inline src/pages/Create.vue]: See previous comment.

File: src/shared/components/OrderList.vue [hunk 4]
```diff
@@ -58,7 +59,8 @@ export default defineComponent({
   },
   methods: {
     handleOrderRowClick(id: string) {
-      alert(id)
+      alert(id);
```
Comment (mllrdev-carlo-medel) [inline src/shared/components/OrderList.vue]: Do you still need this alert?

File: src/core/enums/status.enum.ts (hunk 1 above)
Comment (mllrdev-jerry-bayoneta) [reply src/core/enums/status.enum.ts]: change to CLOSED

File: src/core/services/order.service.ts (hunk 2 above)
Comment (mllrdev-jerry-bayoneta) [reply src/core/services/order.service.ts]: done

File: src/pages/Create.vue (hunk 3 above)
Comment (mllrdev-jerry-bayoneta) [reply src/pages/Create.vue]: done

File: src/shared/components/OrderList.vue (hunk 4 above)
Comment (mllrdev-jerry-bayoneta) [reply src/shared/components/OrderList.vue]: deleted alert(id);

File: src/core/services/order.service.ts [hunk 5]
```diff
@@ -56,11 +56,11 @@ const purchaseStatusUpdate = async(statusUpdate: IChangePurchaseStatus) => {
...
   })
-    .then( response => {
-      console.log(response.data.message);
+    .then(() => {
+      return true
     })
-    .catch( error => {
-      console.log(error)
+    .catch((error) => {
+      return false
```
Comment (mllrdev-carlo-medel) [inline src/core/services/order.service.ts]:
```suggestion
    .catch(() => {
      return false
```

File: src/pages/Create.vue [hunk 6]
```diff
@@ -180,7 +180,17 @@ export default defineComponent({
     },
     declinePurchaseOrder() {
       this.orderStatusUpdate.orderDetails.status = PurchaseStatus.DENIED;
-      orderService.approver.purchaseStatusUpdate(this.orderStatusUpdate);
+      orderService.approver.purchaseStatusUpdate(this.orderStatusUpdate).then((confirm: boolean) => {
+        switch(confirm){
```
Comment (mllrdev-carlo-medel) [inline src/pages/Create.vue]:
```suggestion
      orderService.approver.purchaseStatusUpdate(this.orderStatusUpdate).then((success: boolean) => {
        switch(success){
```

File: src/core/enums/status.enum.ts [hunk 7]
```diff
@@ -2,7 +2,7 @@ enum PurchaseStatus {
     PENDING = 'PENDING',
     APPROVED = 'APPROVED',
     DENIED = 'DENIED',
-    CLOSE = 'CLOSE'
+    CLOSE = 'CLOSED'
```
Comment (mllrdev-carlo-medel) [inline src/core/enums/status.enum.ts]:
```suggestion
    CLOSED = 'CLOSED'
```

File: src/core/services/order.service.ts (hunk 5 above)
Comment (mllrdev-jerry-bayoneta) [reply src/core/services/order.service.ts]: done

File: src/pages/Create.vue (hunk 6 above)
Comment (mllrdev-jerry-bayoneta) [reply src/pages/Create.vue]: done

File: src/core/enums/status.enum.ts (hunk 7 above)
Comment (mllrdev-jerry-bayoneta) [reply src/core/enums/status.enum.ts]: done

**Review by mllrdev-carlo-medel — APPROVED** [review]

**Pull request merged**